### PR TITLE
Show potentially relevant session recordings in funnel persons modal

### LINKED ISSUE #5664: View recording(s) relevant to selected person that dropped off at/completed funnel step
## Is your feature request related to a problem?

Thanks to the persons modal, I'm able to see what people dropped off at/completed funnel step. I have the _identity_ this way, but I don't actually know _why_ they did what they did and in what circumstances. 

Relevant issues:
- #3765
- #5342
- #5545
- #5543

## Describe the solution you'd like

Screen Shot 2021-08-19 at 21 43 01

We have this button "View details" currently, but it works _very_ poorly (#5326) – let's rename to it to "View sessions" and make it work. It should take you to the Sessions page and show sessions in which the funnel was entered and progressed through (possibly multiple since funnels can easily cross session boundaries). Sessions with recordings should have play buttons.

The core issue seems to be that the Sessions page is day-by-day, and many funnels are set up for longer periods, so this limit will have to be lifted. I believe it was put in place to avoid performance problems, but it should be just fine for a single person.

One worry of mine is that the Sessions page experience is a bit weird due to conflating 2 really different pieces of data: calculated sessions and session recordings (see #4884) – hence a session can have multiple play buttons. That's for a different issue though.

Issues to watch out for implementation-wise here:
- #5447 – bug that may or may not cause issues with filtering by person, will see
- #5173 – feature that would be great to get in too, as sessions with recordings are particularly relevant for figuring out the why

Here's a relevant design posted by @clarkus in #5326:

image

It's really interesting, but I think we'll have to go for something slightly different, as:
1. We've dropped the persons count in #5559 to avoid small glaring inconsistencies.
2. A single "Watch" button is not tenable in practice due to problems laid out above and in #4884. I think eventually it will be possible, but for now I'd rather enable this via properly working "View sessions".

## Additional context

Part of "Diagnosing Causes".

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Resolves #5664.

Screen Shot 2021-08-29 at 00 50 40

## Checklist

- [x] All querysets/queries filter by Organization, by Team, and by User
- [x] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by Twixes** [comment]
Ready for review.

This way of correlating relevant recordings is pretty naive, but it works really efficiently. It shows all recordings that were captured between `beginning of funnel time frame` and `end of funnel time frame extended with the funnel window`.

I also attempted ways to correlate this with funnel events, but
1. it's easy to miss a tangentially related, but still relevant recording showing what the user was doing,
2. it's much more expensive to do a `JOIN` like that.

**Comment by macobo** [comment]
> This way of correlating relevant recordings is pretty naive, but it works really efficiently. It shows all recordings that were captured between beginning of funnel time frame and end of funnel time frame extended with the funnel window.

While your rationale makes sense, this causes some issues:
1. We're not able to "jump to $TIME" in session recording that's most relevant
2. There'll be a _lot_ of extra noise in the list of recordings (the time frame can be large)

As-is I don't think this solves the right problem.

Also re JOIN being expensive - AFAIK we'll be doing something extremely similar in paths to find where users went after $STEP. Will quickly sync with @neilkakkar or @EDsCODE whether the approach is applicable here.

Will take a proper look after that discussion. Given you've switched teams it might make sense to merge this even if we'll be changing the approach and someone from core analytics doing a follow-up.

**Comment by Twixes** [comment]
I think this is much better than what we've got currently (nothing), but I agree with both points.
They're both kind of tricky to resolve though:
1. As for jumping to relevant time:
	- A single session recording can easily contain multiple funnel steps being performed – do we show that session recording multiple times with different timestamps, or just take the user to the first timestamp?
2. As for only showing the most relevant recordings:
	- This is pretty straightforward for step completions, since we can only show recordings where the relevant step was performed. What about dropoffs though – how do we show the recording relevant to why the user _didn't_ perform the step? In this case only a human can judge whether the recording is noise in the context or not.

**Review by paolodamico — COMMENTED** [review]
UX-wise & frontend code look good. I'm not the right person to review the CH changes, though I agree with the discussion here, this is better than nothing, yet I don't think it's the right approach just yet. Users will not receive value if we don't show them the relevant part of the recordings easily. Particularly the entire funnel timeframe might be comprised of many sessions & long sessions too, it could be hard to find where users dropped off. Perhaps this will be easier when we combine with quant analysis too.

File: frontend/src/scenes/sessions/SessionRecordingsButton.tsx [hunk 1]
```diff
@@ -13,8 +13,12 @@ interface SessionRecordingsButtonProps {
     sessionRecordings: SessionRecordingType[]
 }
 
-export const sessionPlayerUrl = (sessionRecordingId: string): string => {
-    return `${location.pathname}?${toParams({ ...fromParams(), sessionRecordingId })}`
+export function sessionPlayerUrl(sessionRecordingId: string): string {
+    if (location.pathname.startsWith('/sessions')) {
```
Comment (macobo) [inline frontend/src/scenes/sessions/SessionRecordingsButton.tsx:17]: Pretty sure this breaks navigation on persons page.

**Review by macobo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**